core/src/main/java/bio/terra/pearl/core/dao/survey/AnswerDao.java

```diff
@@ -76,4 +76,20 @@ public Answer findForEnrolleeByQuestion(UUID enrolleeID, String surveyStableId,
                         .orElse(null)
         );
     }
+
+    public List<Answer> findAllForEnrolleeByQuestion(UUID enrolleeID, String surveyStableId, List<String> questionStableId) {
+        return jdbi.withHandle(handle ->
+                handle.createQuery("""
+                                   SELECT DISTINCT ON (question_stable_id) * FROM answer
```

Comment:
can you add a comment reminding the reader how distinct on filters duplicates?  IIRC it grabs the "first", which is why you have it ordered by last_updated_at to get the most recent?

Reply:
Yep, that's exactly it! I'll leave a comment explaining that.